src/Compilers/CSharp/Portable/Compilation/SpeculativeSemanticModelWithMemberModel.cs

```diff
@@ -0,0 +1,565 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
```

Comment:
Consider adding xml doc.
nit: extra blank line below

Reply:
> Consider adding xml doc.

I think the name is descriptive enough and captures everything that the doc comment would say.